domain/src/test/java/org/oppia/android/domain/platformparameter/PlatformParameterControllerTest.kt

```diff
@@ -0,0 +1,260 @@
+
+  @Rule
+  @JvmField
+  val mockitoRule: MockitoRule = MockitoJUnit.rule()
+
+  @Rule
+  @JvmField
+  val executorRule = InstantTaskExecutorRule()
+
+  @Inject
+  lateinit var platformParameterController: PlatformParameterController
+
+  @Inject
+  lateinit var platformParameterSingleton: PlatformParameterSingleton
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+
+  @Mock
+  lateinit var mockUnitObserver: Observer<AsyncResult<Unit>>
+
+  @Captor
+  lateinit var unitCaptor: ArgumentCaptor<AsyncResult<Unit>>
+
+  private val mockPlatformParameterList by lazy {
+    listOf<PlatformParameter>(
+      PlatformParameter.newBuilder().setName(STRING_PLATFORM_PARAMETER_NAME)
+        .setString(STRING_PLATFORM_PARAMETER_VALUE).build(),
+      PlatformParameter.newBuilder().setName(INTEGER_PLATFORM_PARAMETER_NAME)
+        .setInteger(INTEGER_PLATFORM_PARAMETER_VALUE).build(),
+      PlatformParameter.newBuilder().setName(BOOLEAN_PLATFORM_PARAMETER_NAME)
+        .setBoolean(BOOLEAN_PLATFORM_PARAMETER_VALUE).build()
+    )
+  }
+
+  @Test
+  fun testController_noPreviousDatabase_readPlatformParameters_platformParameterMapIsEmpty() {
+    setUpTestApplicationComponent()
+    platformParameterController.getParameterDatabase().toLiveData().observeForever(mockUnitObserver)
+    testCoroutineDispatchers.runCurrent()
+
+    // The platformParameterMap must be empty as there was no previously cached data.
+    verify(mockUnitObserver, atLeastOnce()).onChanged(unitCaptor.capture())
+    assertThat(unitCaptor.value.isSuccess()).isTrue()
+    assertThat(platformParameterSingleton.getPlatformParameterMap()).isEmpty()
+  }
+
+  @Test
+  fun testController_existingDatabase_readPlatformParameters_platformParameterMapHasValues() {
+    // Simulate that previous app already has cached platform parameter values in cache store.
+    executeInPrevious { testComponent ->
+      testComponent.getPlatformParameterController().updatePlatformParameterDatabase(
+        mockPlatformParameterList
+      )
+      testComponent.getTestCoroutineDispatchers().runCurrent()
+    }
+
+    // Create the application after previous arrangement to simulate a re-creation.
+    setUpTestApplicationComponent()
+    platformParameterController.getParameterDatabase().toLiveData().observeForever(mockUnitObserver)
+    testCoroutineDispatchers.runCurrent()
+
+    // The platformParameterMap must have values as application had cached platform parameter data.
+    verify(mockUnitObserver, atLeastOnce()).onChanged(unitCaptor.capture())
+    assertThat(unitCaptor.value.isSuccess()).isTrue()
+    assertThat(platformParameterSingleton.getPlatformParameterMap()).isNotEmpty()
+    checkEntriesInsidePlatformParameterMap(platformParameterSingleton.getPlatformParameterMap())
+  }
+
+  @Test
+  fun testController_noPreviousDatabase_updatePlatformParameterDatabase_readPlatformParameters_platformParameterMapHasValues() { // ktlint-disable max-line-length
```

Comment:
Can probably shorten these to be `testController_updateEmptyDatabase_readPlatformParameters_platformParameterMapHasValues` or `testController_updateExistingDatabase_readPlatformParameters_platformParameterMapHasNewValues` etc. (ditto elsewhere)

Reply:
Done